app/src/main/java/com/naman14/timberx/ui/adapters/SongsAdapter.kt

```diff
@@ -44,11 +44,11 @@ class SongsAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
         return when (viewType) {
-            typeSongHeader -> HeaderViewHolder(DataBindingUtil.inflate<ItemSongsHeaderBinding>(LayoutInflater.from(parent.context),
+            typeSongHeader -> HeaderViewHolder(DataBindingUtil.inflate(LayoutInflater.from(parent.context),
```

Comment:
oh jeez, I'd love a method like `fun ViewGroup.inflateWithBinding(@LayoutRes layoutRes: Int, attachToRoot: Boolean = false) = DataBindingUtil.inflate(LayoutInflater.from(this.context).inflate(layoutRes, this, attachToRoot)` and then replace this with `typeSongItem -> ViewHolder(parent.layoutWithBinding(R.layout.item_songs))` and that's it

Reply:
@Zhuinden me too, but out of scope here

Reply:
Well maybe not - but this PR is about cleaning up existing stuff more than adding new stuff